cfme/common/provider.py

```diff
@@ -153,47 +150,153 @@ def create(self, cancel=False, validate_credentials=True, check_existing=False,
         Returns:
             True if it was created, False if it already existed
         """
-        if check_existing and self.exists:
-            created = False
-        else:
-            created = True
-            logger.info('Setting up provider: %s', self.key)
-            navigate_to(self, 'Add')
-            fill(self.properties_form, self._form_mapping(True, hawkular=False, **self.__dict__))
-            for cred in self.credentials:
-                fill(self.credentials[cred].form, self.credentials[cred],
-                     validate=validate_credentials)
-            self._submit(cancel, self.add_provider_button)
-            if not cancel:
-                flash.assert_message_match('{} Providers "{}" was saved'.format(self.string_name,
-                                                                                self.name))
-        if validate_inventory:
-            self.validate()
+        from cfme.infrastructure.provider import InfraProvider
+        if self.one_of(InfraProvider):
+            if check_existing and self.exists:
+                created = False
+            else:
+                created = True
```

Comment:
I noted this and will change when finish provider conversion.  I don't want to change logic right now  because I made many changes and every small additional change may cause unexpected things.

I remember cases when fix of old defect caused a lot of issues because there were a lot of workarounds relying on this bug.